# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: intl/hyphenation/glue/nsHyphenator.cpp
# Commit: a345e05df151
# Full Hash: a345e05df151563625fe05ec63094b3dae2c9379
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2020-07-07 21:31:11
# Regressor Bug: 1627075
# File Overlap Count: 1
# Description:
#   Bug 1627075 - Route Omnijar requests through StartupCache r=froydnj
#   
#   This should be a relatively straightforward patch. Essentially, we implement
#   a wrapper class (and friends) around nsZipArchive (and friends), which transparently
#   caches entries from the underlying zip archive in the StartupCache. This will break
# ==============================================================================

diff -r 288a67aed661 -r a345e05df151 intl/hyphenation/glue/nsHyphenator.cpp
--- a/intl/hyphenation/glue/nsHyphenator.cpp	Tue Jul 07 04:35:08 2020 +0000
+++ b/intl/hyphenation/glue/nsHyphenator.cpp	Tue Jul 07 04:36:27 2020 +0000
@@ -41,20 +41,11 @@
   if (!file) {
     return nullptr;
   }
-  RefPtr<nsZipArchive> archive = Omnijar::GetReader(file);
+  RefPtr<CacheAwareZipReader> archive = Omnijar::GetReader(file);
   if (archive) {
     nsCString path;
     aJAR->GetJAREntry(path);
-    nsZipItem* item = archive->GetItem(path.get());
-    if (item && item->Compression() == 0 && item->Size() > 0) {
-      // We do NOT own this data, but it won't go away until the omnijar
-      // file is closed during shutdown.
-      const uint8_t* data = archive->GetData(item);
-      if (data) {
-        *aLength = item->Size();
-        return data;
-      }
-    }
+    return archive->GetData(path.get(), aLength);
   }
   return nullptr;
 }